components/mission_key_management/policy.c

```diff
@@ -39,10 +89,29 @@ int policy_add(
 // `nonce` with a proper signature from `hmac_key`.  Returns 1 if the
 // attestation is valid and 0 otherwise.
 static int check_hmac(const uint8_t nonce[NONCE_SIZE],
-        const uint8_t measure[MEASURE_SIZE], const uint8_t hmac[HMAC_SIZE]) {
+        const uint8_t measure[MEASURE_SIZE], const uint8_t hmac[HMAC_SIZE]) 
+/*$
+requires 
+    take Nonce_in  = ArrayOwned_u8(nonce, NONCE_SIZE()); 
+    take Measure_in = ArrayOwned_u8(measure, MEASURE_SIZE()); 
+    take Hmac_in = ArrayOwned_u8(hmac, HMAC_SIZE()); 
+ensures 
+    take Nonce_out  = ArrayOwned_u8(nonce, NONCE_SIZE()); 
+    take Measure_out = ArrayOwned_u8(measure, MEASURE_SIZE()); 
+    take Hmac_out = ArrayOwned_u8(hmac, HMAC_SIZE()); 
+$*/
+{
     uint8_t hmac_message[MEASURE_SIZE + NONCE_SIZE] = {0};
+    // TODO gross hack caused by weird CN type lifting 
+    uint8_t *tmp = hmac_message; 
+
+    // TODO Should be automated 
```

Comment:
@septract please link to an issue, or replace with `NOTE` and maybe some explanation